#119 Added Release Notes for version 1.4.0 (#121)
Inbibo/adonisfx_docs: docs/release_notes.md

# Release Notes

## Version 1.4.0

### Core

> - Implemented a new solver to make muscles behave more compact: AdnGlue solver.

Reviewer: ```suggestion
- Implemented a new solver to glue muscles together so they behave more compact: AdnGlue solver.
```

> - Implemented Soft Constraints.

Reviewer 2: If you are referring to soft constraints in AdnGlue maybe we could say "Implemented Glue Constraints" as this is the name we expose for the user.

Author: Good idea, done!

- Added parallelization to Distance constraints.
- Added parallelization to Shape Preservation constraints.

> - Added Hard Constraints to theAdnFat solver.

Reviewer 2: ```suggestion
- Added Hard Constraints to the AdnFat solver.
```

Author: weird that Google Doc did not highlight this one! thank you, fixed now

> - Added parallelization to Distance constraints.
> - Added parallelization to Shape Preservation constraints.
> - Added Hard Constraints to the AdnFat solver.

Reviewer: Do we use capital for `Constraints`?
```suggestion
- Added parallelization to Distance constraints.
- Added parallelization to Shape Preservation constraints.
- Added Hard constraints to the AdnFat solver.
```

Author: Done

- Added control to the minimum parallelizable chunk size via environment variable `MIN_PARALLEL_SIZE`.

> - Increased the Logger verbosity in Debug level.

Reviewer 2: Should we expose this improvement to the user? I mean, the user won't be able to see the debug logs, right? Maybe we could say "Improved logger verbosity to help with troubleshooting" or something like that.

Author: Mmm the user would be able to see debug logs by changing the environment variable. What we should do, probably, is to explain how to modify it. I will add a section in FAQ for this (not in this issue). If that is ok with you, I would leave this line as it is. Please, let me know!

Reviewer: Agreed, it's an improvement in my opinion

Reviewer 2: Okay, makes totally sense, thank you!

### Maya

- Added AdnGlue node.
- Added AdnActivation node.
- Added support to unidirectional muscle to muscle constraints.
- Added support to Hard Constraints in AdnFat solver including paintable map and stiffness override.
- Optimized AdnFat deformer.
- Optimized AdnSkin deformer.
- Optimized AdonisFX Paint Tool.
- Added color interpolation to the fibers debugger in AdnMuscle and AdnRibbonMuscle to encode the muscle activation.
- Deprecated AdnWeightsDisplayNode.
- Added a shortcut in the AdonisFX menu to upgrade deprecated nodes.
- Added the ability to choose which constraints reinitialize at start time.

> ### Bug Fixes

Reviewer 2: Should we add this bugfix?: "1504 Avoid viewport refresh on deformer creation"

Author: I thought about it but I decided to not expose it. It was listed in the Release 1.4.0 card just for us to keep track of it internally. But actaully, it is not really but fix, I mean, nothing was "broken". The removal of that was to avoid the warning in AdnSkin saying that it could not attach to the Debugger. I

Reviewer 2: Got it, thanks for the clarification.

Reviewer: Agreed!
